x-pack/plugins/security_solution/server/search_strategy/security_solution/index.ts

```diff
@@ -19,15 +20,16 @@ export const securitySolutionSearchStrategyProvider = <T extends FactoryQueryTyp
   const es = data.search.getSearchStrategy('es');
 
   return {
-    search: async (context, request, options) => {
+    search: (request, options, context) => {
       if (request.factoryQueryType == null) {
         throw new Error('factoryQueryType is required');
       }
       const queryFactory: SecuritySolutionFactory<T> =
         securitySolutionFactory[request.factoryQueryType];
       const dsl = queryFactory.buildDsl(request);
-      const esSearchRes = await es.search(context, { ...request, params: dsl }, options);
-      return queryFactory.parse(request, esSearchRes);
+      return es
+        .search({ ...request, params: dsl }, options, context)
+        .pipe(mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes)));
```

Comment:
I think you can use `map` instead of `mergeMap` here. Am I missing something?

Reply:
we should use `mergeMap` with promises. In case of `map` it returns unresolved promise
https://www.learnrxjs.io/learn-rxjs/operators/transformation/mergemap

https://stackblitz.com/edit/typescript-pnnsrq?file=index.ts&devtoolsheight=100